Electricitymarket

rajeev-datta opened the pull request:
I made an initial draft of the single-agent problem without a market operator.

rajeev-datta committed:
Add files via upload

rajeev-datta committed:
Add files via upload

Commit:
updated battery storage env

Commit:
updated battery storage env

Commit:
Merge branch 'electricitymarket' of https://github.com/chrisyeh96/sustaingym into electricitymarket

Commit:
small changes and data analysis code

Commit:
small changes and data analysis code

Commit:
Merge branch 'electricitymarket' of https://github.com/chrisyeh96/sustaingym into electricitymarket

Commit:
added data folder

Commit:
updated data reading code

Commit:
testing environment

chrisyeh96 reviewed (changes requested):
In general, if you're committing Jupyter notebooks, please "clear all outputs" before committing them to the repo.

For the pricing data CSV, commit only a processed version, not the raw file. (Write a script that extracts only the necessary columns)

Commit:
replacing data file

Commit:
fixed battery env

Commit:
cleared outputs

Commit:
fixed env and wrote ddpg code

Commit:
ran ddpg code

Commit:
bar graph code

Commit:
more data analysis

Commit:
remove ignored model files

Commit:
remove ignored model files

Commit:
remove ignored model files

Commit:
remove ignored model files

Commit:
renaming files

Commit:
wrotee constructors

Commit:
adding MO problem to env

Commit:
first draft of env

Commit:
debugging

Commit:
debugging env

chrisyeh96 committed:
temp edits by Chris

Commit:
check env runs

Commit:
random actions

Commit:
debugging single agent

Commit:
infeasible constraints

chrisyeh96 committed:
Simplify battery gym

Commit:
debugging tests

Commit:
fixed tests

Commit:
changing reset

Commit:
added costs assurances

Commit:
running PPO on environment

Commit:
including load data

Commit:
random actions work with load data

Commit:
analyzing carbon data and using net demand

chrisyeh96 committed:
add requests to env.yml

Commit:
retrieving moer values

Commit:
Merge branch 'electricitymarket' of https://github.com/chrisyeh96/sustaingym into electricitymarket

Commit:
calculated offline optimal

Commit:
making env more realistic

Commit:
random actions running on more realistic environment w/ forecasts

Commit:
made data plots

Commit:
trying to find dist shift

Commit:
trying to find dist shift

chrisyeh96 committed:
Clean up CAISO demand + forecast data

chrisyeh96 committed:
Chris cleanup

chrisyeh96 committed:
Cleaner variables

Commit:
cleaned up code... will not verify seed logic and run PPO on May 2020

Commit:
debugging out of dist evaluation

Commit:
ran PPO and A2C

chrisyeh96 committed:
vectorize battery charge update

chrisyeh96 committed:
minor updates

chrisyeh96 committed:
use self.obs

chrisyeh96 committed:
fix mypy errors

Commit:
fixed environment

Commit:
fixed environment

Commit:
fixed environment

Commit:
made some plots

Commit:
added demo

Commit:
Merge remote-tracking branch 'origin/main' into electricitymarket

Commit:
changed solver to mosek

chrisyeh96 committed:
Create battery folder

chrisyeh96 committed:
break out reward components

Commit:
fixed environment

Commit:
fixed environment

Commit:
added terminal reward

Commit:
Merge remote-tracking branch 'origin/main' into electricitymarket

Commit:
use MOER loader

Commit:
fixed random actions w/ terminal reward

Commit:
changed training scripts

Commit:
making plots

Commit:
finished presentation plots

Commit:
no change

Commit:
moved files and plotted prices and state of charge

Commit:
added panda power to RL env

Commit:
adding plot helper fxn

Commit:
factoring out plotting code

Commit:
added reward plots

Commit:
changed reward plots and printed soc to debug

Commit:
debugged env and retraining models

Commit:
debugged env and retraining models

Commit:
debugged env and retraining models

Commit:
trained models

Commit:
remade graphs

Commit:
fixed per step offline reward

Commit:
fixing plots

Commit:
fixed  plots

Commit:
retraining with added penalty

Commit:
retraining with added penalty

chrisyeh96 committed:
Merge

Commit:
updated offline optimal code

chrisyeh96 committed:
refactor offline optimal

Commit:
trained models with terminal cost

Commit:
updated results with terminal cost

Commit:
trained with increased gamma

Commit:
rescaling action space

Commit:
tracked actions taken during training

Commit:
tracked actions taken during training

Commit:
added logging action callback

chrisyeh96 committed:
Add discrete action space to electricity market

chrisyeh96 committed:
Cleanup

Commit:
running 250k training steps

Commit:
ran models on discrete action space

Commit:
running optimal vs discrete

chrisyeh96 committed:
Merge branch 'main' into electricitymarket

chrisyeh96 committed:
Document generators + battery system parameters

Commit:
ran dqn on discrete action space

Commit:
Merge branch 'electricitymarket' of https://github.com/chrisyeh96/sustaingym into electricitymarket

chrisyeh96 committed:
Update actions to be (charge cost, discharge cost)

Also use wrapper for discrete actions
General code refactoring
Better documentation

chrisyeh96 committed:
Update charging and discharging efficiency

chrisyeh96 committed:
Fix approx offline optimal to account for charging efficiency

not tested yet

chrisyeh96 committed:
Use 3-action discretization

Commit:
added bar graph and debugged

Commit:
updated offline optimal plotting code

chrisyeh96 committed:
Fix offline optimal

Set roundtrip efficiency to ~90% (matches Tesla Powerwall). Lower values
lead to optimal agent doing nothing.

chrisyeh96 committed:
Update plot utils

chrisyeh96 committed:
Updates

chrisyeh96 committed:
Move price to state

chrisyeh96 committed:
Return offline opt results via dict

Fix mypy + flake8 issues

chrisyeh96 committed:
Typo

chrisyeh96 committed:
update typing annotations

chrisyeh96 committed:
Enable setting bats_init_energy

Commit:
added follow optimal plotting code

chrisyeh96 committed:
Fix plot_utils

chrisyeh96 committed:
Unify training code

chrisyeh96 committed:
Update data plots

Commit:
ran models... debugging plotting code

chrisyeh96 committed:
Many updates

- option to not use intermediate rewards
- simplify plotting code
- combine training code

chrisyeh96 committed:
More significant changes

- train.py now support --intermediate-rewards
- fixed bugs in run_electricitymarket.py
- now using seaborn to make violinplots

chrisyeh96 committed:
analysis notebooks

chrisyeh96 committed:
Figures

Commit:
cleaning up...

Commit:
cleaning up

Commit:
refactoring test code and cleaning up

Commit:
altered training script and revisted plotting script

Commit:
updated plot code and notebooks

Commit:
ignoring model_analysis.ipynb file

rajeev-datta committed:
Update electricity_market.py
